### PR TITLE
RFC 77, RFC 264: CloudFormation Registry Imports

### PR DESCRIPTION
**Pull request by rix0rrr**
These are two distinct RFCs for two distinct features, but we might as
well work on the docs at the same time. Hence, one PR.

* Tracking issue: #77 - importing external resources
* [Rendered version](/huijbers/0264-0077-cloudformation-registry/text/0077-import-external-resources.md)

* Tracking issue: #264 - codegen off the new registry schema
* [Rendered version](/huijbers/0264-0077-cloudformation-registry/text/0264-registry-schema-codegen.md)

---

_By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache-2.0 license_

### CODE REVIEW

File: text/0264-registry-schema-codegen.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+  our build. Since a number of resources are automatically being converted
+  from old format to new format, they are missing additional information. We
+  can enhance the spec with information obtained from the [cfn-lint] project.
+* A tool to two versions of the spec. This is necessary to generate the
+  changelog every time we import a new version of the resources. It should be
+  able to identify breaking changes, so that we can extend this tool to
+  resource developers and they can prevent themselves from accidentally
+  releasing breaking changes (reducing churn on downstream systems).
+* A tool to generate code the type and resource definitions found
+  in the schema. This can mostly be based on `json2jsii`, although we will
```
Comment (eladb) [inline text/0264-registry-schema-codegen.md:144]: Let's definitely try to bake as much of the capabilities into Json2jsii so we get the most leverage for other CDKs (and jsii-based projects).

**Review by eladb — CHANGES_REQUESTED** [review]
Great RFCs! Let's definitely have a discussion as part of these two RFCs around whether we want to continue to vend CFN resources as part of the AWS construct library or just use them as an implementation detail.

**Review by eladb — CHANGES_REQUESTED** [review]
We should make sure to take the recently launched CloudFormation modules into the `cdk import` feature. I'd like to be able to do something like `cdk import MyCompany::S3::Bucket::MODULE`

**Comment by nija-at** [comment]
@rix0rrr - can you check if this is superceded by https://github.com/aws/aws-cdk-rfcs/pull/356.

**Comment by rix0rrr** [comment]
> @rix0rrr - can you check if this is superceded by #356.

From the title, #356 would build on the work proposed here, but the work here would also be useful to people importing constructs in their own accounts.

**Comment by eladb** [comment]
FYI, this is now being implemented and published under [cdklabs/cdk-import](https://github.com/cdklabs/cdk-import).

@rix0rrr I am wondering if it's worth leaving this RFC open or we can merge it for now and continue iterating on the code?

**Pull request merged**